### PR TITLE
[COR-553] Added ANY/ALL/NONE support for LIKE operator

### PR DESCRIPTION
**Pull request by bluepal-avanthi-dundigala**
- Added array quantifier support for LIKE and NOT LIKE expressions, enabling syntax such as:
        ["foo", "bar"] ANY LIKE "b%"
        ["foo", "bar"] AT LEAST(2) LIKE "b%"

- Implemented parse time desugaring to the existing boolean array expansion form:
        array[? quantifier FILTER CURRENT LIKE pattern]
        via Ast::createNodeArrayLikeOperator
        This reuses the existing execution infrastructure and avoids introducing a new executor path.

- Extended the lexer and parser with compound tokens (T_ANY_LIKE, T_NOT_LIKE, etc.) and added the necessary disambiguation logic to ensure correct parsing of:
        Graph traversal ANY
        Scalar NOT LIKE
        LIKE(...) function calls

- Added parser coverage in aql-parse.js.

- Added runtime validation tests in aql-array-like.js.

Fixes: https://github.com/arangodb/arangodb/issues/10372

### CODE REVIEW

File: CHANGELOG
Comment (bluepal-pavani-lanka) [inline CHANGELOG:1]: Minor change, cor-553 id is missing in CHANGELOG entry

**Comment by bluepal-pavani-lanka** [comment]
@bluepal-avanthi-dundigala  few file changes from COR-303 are showing as changed files in this PR. Please have a look

**Comment by bluepal-pavani-lanka** [comment]
Try to cover few more test scenarios:

1. Invalid Case: 
'RETURN ["foo"] ANY LIKE'
'RETURN ["foo"] AT LEAST() LIKE "x%"'

2. At Least zero LIKE:
db._query('RETURN @values AT LEAST(0) LIKE "x%"', { values: ["foo", "bar", "baz"]  }).toArray()

3. At Least One LIKE:
db._query('RETURN @values AT LEAST(4) LIKE "f%"', { values: ["foo", "bar", "baz"] }).toArray());

4. Corner Cases:
RETURN [""] ANY LIKE ""
RETURN ["foo"] ANY LIKE ""

5. Added tests for empty values also Test with Null values :
RETURN [null, "bar"] ANY LIKE "b%"
RETURN [null] ALL LIKE "b%"
RETURN [null] NONE LIKE "b%"

6. Non-String values:
RETURN [1, 2, 3] ANY LIKE "1%"
RETURN [true, false] ANY LIKE "t%"

7. RETURN null ANY LIKE "f%"
RETURN 123 ANY LIKE "1%"
RETURN {} ANY LIKE "%"

8. Escape character support:
RETURN ["a_b"] ANY LIKE "a\\_b"
RETURN ["a%b"] ANY LIKE "a\\%b"

db._query('RETURN [ 'möt', 'ör', 'möy' ] ALL LIKE "mö%"').toArray();

File: arangod/Aql/Ast.cpp [hunk 1]
```diff
@@ -1337,6 +1337,38 @@ AstNode* Ast::createNodeBooleanExpansion(int64_t levels,
...
+                                          bool negate) {
+  TRI_ASSERT(lhs != nullptr);
+  TRI_ASSERT(pattern != nullptr);
+  TRI_ASSERT(quantifier != nullptr);
+
+  std::string const varName = variables()->nextName() + "_";
+  AstNode* iterator =
+      createNodeIterator(varName.c_str(), varName.size(), lhs);
+  auto* variableNode = iterator->getMember(0);
+  TRI_ASSERT(variableNode->type == NODE_TYPE_VARIABLE);
```
Comment (bluepal-pavani-lanka) [inline arangod/Aql/Ast.cpp:1353]: TRI_ASSERT(iterator->numMembers() >= 1);
asserting the iterator before accessing member 0.

**Review by bluepal-pavani-lanka — COMMENTED** [review]
Added minor comments..

**Comment by bluepal-avanthi-dundigala** [comment]
> @bluepal-avanthi-dundigala few file changes from COR-303 are showing as changed files in this PR. Please have a look

I resolved it by removing COR-303 files

**Comment by bluepal-avanthi-dundigala** [comment]
> Try to cover few more test scenarios:
> 
> 1. Invalid Case:
>    'RETURN ["foo"] ANY LIKE'
>    'RETURN ["foo"] AT LEAST() LIKE "x%"'
> 2. At Least zero LIKE:
>    db._query('RETURN @values AT LEAST(0) LIKE "x%"', { values: ["foo", "bar", "baz"]  }).toArray()
> 3. At Least One LIKE:
>    db._query('RETURN @values AT LEAST(4) LIKE "f%"', { values: ["foo", "bar", "baz"] }).toArray());
> 4. Corner Cases:
>    RETURN [""] ANY LIKE ""
>    RETURN ["foo"] ANY LIKE ""
> 5. Added tests for empty values also Test with Null values :
>    RETURN [null, "bar"] ANY LIKE "b%"
>    RETURN [null] ALL LIKE "b%"
>    RETURN [null] NONE LIKE "b%"
> 6. Non-String values:
>    RETURN [1, 2, 3] ANY LIKE "1%"
>    RETURN [true, false] ANY LIKE "t%"
> 7. RETURN null ANY LIKE "f%"
>    RETURN 123 ANY LIKE "1%"
>    RETURN {} ANY LIKE "%"
> 8. Escape character support:
>    RETURN ["a_b"] ANY LIKE "a\_b"
>    RETURN ["a%b"] ANY LIKE "a\%b"
> 
> db._query('RETURN [ 'möt', 'ör', 'möy' ] ALL LIKE "mö%"').toArray();

Locked tests cases

**Review by maierlars — APPROVED** [review]
